### PR TITLE
clarify java 17 search docs

### PR DESCRIPTION
**Pull request by dshimo**
[Preview](https://github.com/dshimo/liberty-tools-vscode/blob/settingsDocs/docs/user-guide.md#settings)

### CODE REVIEW

File: docs/user-guide.md [hunk 1]
```diff
@@ -38,7 +38,13 @@ This guide provides detailed instructions on how to configure your Liberty proje
 
 ### Settings
 
-To ensure that the Liberty Config Language Server starts properly, define the `xml.java.home` property in your VS Code settings or set the system `JAVA_HOME` environment variable to point to Java 17 .
+Liberty Tools will go through a series of checks to locate Java 17 and will alert you when it fails to do so.
```
Comment (TrevCraw) [inline docs/user-guide.md:41]: what about higher than Java 17?

File: docs/user-guide.md [hunk 2]
```diff
@@ -38,7 +38,13 @@ This guide provides detailed instructions on how to configure your Liberty proje
...
+
+Liberty Tools first checks the [embedded JRE](https://github.com/redhat-developer/vscode-java#java-tooling-jdk) from the `redhat.java` plugin. If handling multiple Java runtimes causes this requirement to fail, you can define properties in your VS Code [settings.json](https://code.visualstudio.com/docs/getstarted/settings) to help point Liberty Tools to your Java 17 runtime. Liberty Tools will search for `java.jdt.ls.java.home` and `java.home` before checking for the system environment variables `JDK_HOME` and `JAVA_HOME`.
+
+Liberty Config Language Server support for XML does not go through this same process, so it's recommended users define `xml.java.home` to ensure full Liberty Tools functionality.
```
Comment (TrevCraw) [inline docs/user-guide.md:45]:
```suggestion
Liberty Config Language Server support for XML does not go through this same process, so it's recommended users define `xml.java.home` to ensure full Liberty config language support.
```

File: docs/user-guide.md [hunk 3]
```diff
@@ -38,7 +38,13 @@ This guide provides detailed instructions on how to configure your Liberty proje
 
 ### Settings
 
-To ensure that the Liberty Config Language Server starts properly, define the `xml.java.home` property in your VS Code settings or set the system `JAVA_HOME` environment variable to point to Java 17 .
+Liberty Tools will go through a series of checks to locate Java 17 and will alert you when it fails to do so.
+
+Liberty Tools first checks the [embedded JRE](https://github.com/redhat-developer/vscode-java#java-tooling-jdk) from the `redhat.java` plugin. If handling multiple Java runtimes causes this requirement to fail, you can define properties in your VS Code [settings.json](https://code.visualstudio.com/docs/getstarted/settings) to help point Liberty Tools to your Java 17 runtime. Liberty Tools will search for `java.jdt.ls.java.home` and `java.home` before checking for the system environment variables `JDK_HOME` and `JAVA_HOME`.
```
Comment (TrevCraw) [inline docs/user-guide.md:43]: "If handling multiple Java runtimes causes this requirement to fail" - I'm not sure I follow what this means

Comment (TrevCraw) [inline docs/user-guide.md:43]: Just noting we may need to mention "Java 17 or later" here

Comment (TrevCraw) [inline docs/user-guide.md:43]: Looks like `java.home` is deprecated  -https://github.com/redhat-developer/vscode-java#supported-vs-code-settings. Should we avoid mentioning it?

Comment (dshimo) [reply docs/user-guide.md]: I've now avoided mentioning it as a main suggestion but included it the comprehensive list of values we check in our requirements.ts

**Review by dmuelle — CHANGES_REQUESTED** [review]
ID review

File: docs/user-guide.md [hunk 4]
```diff
@@ -38,7 +38,20 @@ This guide provides detailed instructions on how to configure your Liberty proje
 
 ### Settings
 
-To ensure that the Liberty Config Language Server starts properly, define the `xml.java.home` property in your VS Code settings or set the system `JAVA_HOME` environment variable to point to Java 17 .
+Liberty Tools for Visual Studio Code requires Java 17 or later to ensure that the [Liberty Config Language Server](https://github.com/OpenLiberty/liberty-language-server) and [Eclipse Language Server for Jakarta EE](https://github.com/eclipse/lsp4jakarta) starts properly. A toast message alerts you if the Liberty Language Config server fails to run.
```
Comment (TrevCraw) [inline docs/user-guide.md:41]:
```suggestion
Liberty Tools for Visual Studio Code requires Java 17 or later to ensure that the [Liberty Config Language Server](https://github.com/OpenLiberty/liberty-language-server) and the [Eclipse Language Server for Jakarta EE](https://github.com/eclipse/lsp4jakarta) start properly. A toast message alerts you if the Liberty Language Config server fails to run.
```

Comment (TrevCraw) [reply docs/user-guide.md:41]: @dmuelle does this suggestion look OK to you?

File: docs/user-guide.md [hunk 5]
```diff
@@ -38,7 +38,20 @@ This guide provides detailed instructions on how to configure your Liberty proje
 
 ### Settings
 
-To ensure that the Liberty Config Language Server starts properly, define the `xml.java.home` property in your VS Code settings or set the system `JAVA_HOME` environment variable to point to Java 17 .
+Liberty Tools for Visual Studio Code requires Java 17 or later to ensure that the [Liberty Config Language Server](https://github.com/OpenLiberty/liberty-language-server) and the [Eclipse Language Server for Jakarta EE](https://github.com/eclipse/lsp4jakarta) start properly. A toast message alerts you if the Liberty Language Config server fails to run.
```
Comment (TrevCraw) [inline docs/user-guide.md:41]: Use Liberty Config Language Server in last sentence too.

Also, is there a toast message if LSP4Jakarta fails to run?

Comment (TrevCraw) [inline docs/user-guide.md:41]: Use Liberty Config Language Server in last sentence too.

Also, is there a toast message if LSP4Jakarta fails to run?

Comment (dshimo) [reply docs/user-guide.md:41]: Ah. This is the clarification.
The check in Liberty Tools for VS Code is for both LSP4Jakarta and LCLS. The compliant JDK is used to start both servers. So instead, maybe the toast should say Liberty Tools for VS Code requires Java 17.
We do not have an alert from our plugin if Lemminx fails for LCLS. That is an entirely separate issue.

Comment (TrevCraw) [reply docs/user-guide.md:41]: So maybe the last sentence should be: "A toast message alerts you if the either of the language servers fail to run."

And then we should open an issue to update the toast message

**Comment by dmuelle** [comment]
per @evie-lau  [comment on the Intellij side](https://github.com/OpenLiberty/liberty-tools-intellij/pull/450#issuecomment-1593592653), does any of this info need to be duplicated for Java 17 statements in the README? Or you might just add a link from README support statement to this doc section for more info. Whatever the case, would be good to handle it the same way in both docs

**Comment by dshimo** [comment]
> per @evie-lau [comment on the Intellij side](https://github.com/OpenLiberty/liberty-tools-intellij/pull/450#issuecomment-1593592653), does any of this info need to be duplicated for Java 17 statements in the README? Or you might just add a link from README support statement to this doc section for more info. Whatever the case, would be good to handle it the same way in both docs

We could include a note or screenshot of the runtime check in #requirements and then link out our provided solution for more info. This way, IntelliJ and VSCode could echo a similar README structure and have their specific solutions to their own user-guide

**Comment by TrevCraw** [comment]
@dmuelle The original reason for why the Java 17 requirements are duplicated was because it was not clear to our users. They reported not knowing they needed Java 17 even though we specified in the user guide. So the README was updated to be explicit about requirements up front.

That being said, I'm open to suggestions on how to better organize this info so that we are clear about the requirements in the README and limit the amount of repeated info - whether that means having basic requirements in the README and more details on the user guide (I think this is along the lines of what @dshimo is suggesting)

**Comment by dmuelle** [comment]
@TrevCraw @dshimo - I think it makes sense to mention Java requirements up front in REAMDE/quickstart as not all users will otherwise need to refer to the user guide. But keeping that information as minimal as possible and then linking to the user guide for context/details is the best strategy from a maintenance perspective. 

We'll still have two sources to update when support levels change, but that's not especially burdensome. We might even add a comment to the MD sources to remind doc maintainers to update both references if support levels change

**Comment by TrevCraw** [comment]
@dmuelle that sounds good. We're going to open a new PR with formatting changes for duplicate info between README and the user guide that aligns VS Code and IntelliJ

**Review by TrevCraw — APPROVED** [review]

**Pull request merged**